README.md

```diff
@@ -15,3 +15,185 @@ To run the tests:
 ```bash
 yarn test
 ```
+
+## Concepts
+
+`directo` is structured as parser for English imperative sentences. All commands must start with a `verb`. Recognized command structures are:
+
+- `verb`
+  - login
+  - exit
+- `verb object`
+  - go north
+  - buy sword
+  - get lamp
+- `verb preposition object`
+  - sit on the log
+- `verb object preposition subject`
+  - give sword to steve
+- `verb determiner object`
+  - eat the berries _Many of these are likely the same as `verb object`, but quantifiers make a difference here._
+- `verb subject determiner object`
+  - give steve the sword
+  - give steve a mushroom
+
+When using `directo`, commands are declared based on the set of verbs and command structure they accept. Many command implementations will want to handle multiple structures (see the two ways one could give a sword to steve above).
+
+## Usage
+
+Using `directo` is as simple as registering commands and handlers:
+
+```javascript
+const { command, processText, Format, CommandResult } = require('directo');
+
+command({
+  verb: 'go',
+  func: async ({ format, object }) => {
+    if (format !== Format.VO) {
+      console.log('Go where?');
+      return CommandResult.FORMAT_ERROR;
+    }
+
+    console.log(`Okay, I went ${object}`);
+
+    return CommandResult.HANDLED;
+  }
+});
+
+(async (commands) => {
+  for (const c of commands) {
+    await processText(c);
+  }
+})([
+  'go north',
+  'go south',
+  'go'
+]);
+```
+
+Prints:
+
+```none
```

Comment:
Thanks for knowing to use `none` as the format here. A lot of people I work with seem to be lacking in Markdown knowledge and often don't note the type at all. ❤️